WooCommerce/Classes/ViewRelated/Dashboard/Settings/In-Person Payments/InPersonPaymentsMenuViewModel.swift

```diff
@@ -0,0 +1,89 @@
+import Foundation
+import SwiftUI
+import Yosemite
+import WooFoundation
+
+class InPersonPaymentsMenuViewModel: ObservableObject {
+    @Published private(set) var shouldShowTapToPaySection: Bool = true
+    @Published private(set) var shouldShowCardReaderSection: Bool = true
+    @Published private(set) var setUpTryOutTapToPayRowTitle: String = Localization.setUpTapToPayOnIPhoneRowTitle
+    @Published private(set) var shouldShowTapToPayFeedbackRow: Bool = true
+    @Published private(set) var shouldDisableManageCardReaders: Bool = false
+    var shouldAlwaysHideSetUpButtonOnAboutTapToPay: Bool = false
+
+    let siteID: Int64
+
+    struct Dependencies {
+        let cardPresentPaymentsConfiguration: CardPresentPaymentsConfiguration
+        let onboardingUseCase: CardPresentPaymentsOnboardingUseCaseProtocol
+        let cardReaderSupportDeterminer: CardReaderSupportDetermining
+    }
+
+    let dependencies: Dependencies
+
+    init(siteID: Int64,
+         dependencies: Dependencies) {
+        self.siteID = siteID
+        self.dependencies = dependencies
+        updateOutputProperties()
+    }
+
+    func updateOutputProperties() {
+        Task {
+            await shouldAlwaysHideSetUpButtonOnAboutTapToPay = dependencies.cardReaderSupportDeterminer.hasPreviousTapToPayUsage()
+        }
+    }
+
+    var setUpTapToPayViewModelsAndViews: SetUpTapToPayViewModelsOrderedList {
```

Comment:
For these variables, is it convenient that we create new objects each time we call them, or can we make them lazy?

Reply:
Great spot, this does cause issues – they're all lazy in a later PR :)